src/time_evolution/time_evolution.jl

```diff
@@ -212,6 +212,19 @@ struct DiscreteLindbladJumpCallback <: LindbladJumpCallbackType end
 
 ContinuousLindbladJumpCallback(; interp_points::Int = 10) = ContinuousLindbladJumpCallback(interp_points)
 
+function _check_tlist(tlist, T::Type)
+    tlist2 = convert(Vector{T}, tlist) # Convert it to support GPUs and avoid type instabilities for OrdinaryDiffEq.jl
+
+    # Check if the list of times is not empty
+    isempty(tlist2) && throw(ArgumentError("The list of times must not be empty."))
+    # Check if the list of times is sorted
+    !issorted(tlist2) && throw(ArgumentError("The list of times must be sorted."))
+    # Check if the list of times is unique
+    length(tlist2) != length(unique(tlist2)) && throw(ArgumentError("The list of times must be unique."))
```

Comment:
how about we just use `allunique`, like:
```suggestion
    allunique(tlist2) || throw(ArgumentError("The time list must be unique."))
```

In this case, I think it won't need extra allocations ?